include/flox/core/exceptions.hh

```diff
@@ -74,76 +74,150 @@ enum error_category {
 /** Typed exception wrapper used for misc errors. */
 class FloxException : public std::exception
 {
-private:
 
-  /** Corresponds to an error_category, in this case `EC_FLOX_EXCEPTION`. */
-  static constexpr std::string_view categoryMsg = "general error";
+private:
 
   /** Additional context added when the error is thrown. */
   std::optional<std::string> contextMsg;
 
   /**
-   * If some other exception was caught before throwing this one, caughtMsg
+   * If some other exception was caught before throwing this one, @a caughtMsg
    * contains what() of that exception.
    */
   std::optional<std::string> caughtMsg;
 
+  /** The final what() message. */
+  std::string whatMsg;
+
+
 public:
 
   explicit FloxException( std::string_view contextMsg )
```

Comment:
question nonblocking: so these shouldn't be called by children classes because it uses this->getCategoryMessage()? Maybe add a comment to that effect?

Reply:
The child classes need to _redeclare_ constructors.
So for child classes they are expected to use the `FloxException( std::string_view categoryMsg, std::optional<std::string> contextMsg, std::optional<std::string> caughtMsg )` CTX.

Reply:
(nonblocking)
I was asking if
```
ChildException( std::string_view contextMsg) : FloxException(contextMsg)
```
would be a bug, and if so to maybe add a comment here to make that clear

Reply:
Yeah it will cause the message to be `generic error: XXX` which likely isn't what the caller expects.
I can add inline comments.

Reply:
I added notes about which constructors are appropriate where.